### DISCUSSION TITLE
Query Arc Beans with Amazon Lambda Extension

### DISCUSSION THREAD

**Discussion by gilday** [quarkusio/quarkus#32741] [Q&A] [answered]
Hello, I am using the quarkus-amazon-lambda extension to build a `RequestHandler<ScheduledEvent, Void>` handler. I'm running this Lambda handler project in dev mode, but I'm unable to use the endpoint `http://localhost:8080/q/arc/beans` to query the beans that are available.

The dev server appears to be running

> "loggerName":"io.quarkus.amazon.lambda.runtime.AbstractLambdaPollLoop","level":"INFO","message":"Listening on: http://localhost:8080/_lambda_/2018-06-01/runtime/invocation/next","threadName":"Lambda Thread (DEVELOPMENT)"

But it responds to the `/q/arc/beans` request with an HTTP 405.

Are the `/q/` endpoints only available to HTTP projects e.g. Lambda handlers that use RESTEasy Reactive and JAX-RS?

[a comment by an automated account is not shown]

**Comment by bulldog98** [comment, marked as answer]
As discussed in https://github.com/quarkusio/code.quarkus.io/issues/1173 you need vertx-http for the dev ui to work

**Reply by gilday** [reply]
Thanks. Is it common / advisable to include `vertx-http` only for dev and omit this extension from the artifact that is ultimately deployed to AWS Lambda?

**Reply by bulldog98** [reply]
I do not have experience with AWS Lambda, so I cannot really answer that, but I can tell you that is what I do for a process that communicates only via rabbitmq.

**Comment by patriot1burke** [comment]
There's a long standing issue of using the dev ui with our lambda extension.  Basically requires a rewrite of the dev ui plumbing which is why it has been on the back burner for awhile

**Comment by gilday** [comment]
For Gradle users trying to figure out how to exclude the lambda extension from dev while still including it in tests and the production artifact, the following worked for me:

```kotlin
dependencies {
    implementation(enforcedPlatform("io.quarkus:quarkus-bom:3.0.1.Final"))
    implementation("jakarta.ws.rs:jakarta.ws.rs-api")

    runtimeOnly("io.quarkus:quarkus-resteasy")
    runtimeOnly("io.quarkus:quarkus-resteasy-jackson")

    quarkusDev("io.quarkus:quarkus-smallrye-openapi")

    // add lambda extension to production artifact
    quarkusProdRuntimeClasspathConfigurationDeployment("io.quarkus:quarkus-amazon-lambda-rest")

    // also add lambda extension to the test class path for QuarkusTest
    testImplementation("io.quarkus:quarkus-amazon-lambda-rest")
    testImplementation("io.quarkus:quarkus-junit5")
    testImplementation("io.rest-assured:rest-assured")
}
```

**Reply by gilday** [reply]
Disregard: this unblocks dev mode, but `quarkusBuild` doesn't see the Lambda extension, so no Lambda code distribution zip is built.

If any users find a way to add a dependency to `quarkusBuild` but not include that dependency in `quarkusDev`, that would work. I haven't yet figured that out.
